carrier_shipping_label_template/models/report.py

```diff
@@ -0,0 +1,69 @@
+import base64
+import io
+
+from odoo import api, models
+
+
+class IrActionsReportReportlab(models.Model):
+    _inherit = "ir.actions.report"
+
+    def _render_qweb_pdf(self, res_ids=None, data=None):
+        if self.report_name in [
+            "carrier_shipping_label_template.report_shipping_label"
+        ]:
+            self = self.with_context(shipping_label_res_ids=res_ids)
+        return super(IrActionsReportReportlab, self)._render_qweb_pdf(res_ids, data)
+
+    @api.model
+    def _render_qweb_html(self, docids, data=None):
+        """This method generates and returns html version of a report."""
+        if (
+            self.report_name
+            not in ["carrier_shipping_label_template.report_shipping_label"]
+        ) or not docids:
+            return super(IrActionsReportReportlab, self)._render_qweb_html(docids, data)
+        new_doc_ids = []
+        for picking in self.env["stock.picking"].browse(docids):
```

Comment:
Just a pointer:

line 25 to 30 can be skipped by just using search with proper parameters instead of browse

Reply:
@nikunjantala Could you improve this accordingly?

Reply:
@alihashaam I have fixed code in 25 to 30 line.